pkg/skaffold/build/cache/lookup.go

```diff
@@ -131,28 +131,29 @@ func (c *cache) lookupRemote(ctx context.Context, hash, tag string, platforms []
 
 		if !cacheHit {
 			log.Entry(ctx).Debugf("Added digest for %s to cache entry", tag)
-			cachedEntry.Digest = remoteDigest
 			c.cacheMutex.Lock()
-			c.artifactCache[hash] = cachedEntry
+			c.artifactCache[hash] = ImageDetails{Digest: remoteDigest}
```

Comment:
I think skaffold saves digest to cache by checking the remote image after build, the code here might not be necessary and if this works as the code indicates it's confusing
 - we find a remoteDigest with the target tag 
 - in cache miss case, we save the digest, skaffold will rebuild and re-push,  then the digest changes on remote but the digest on the cache is old one??

Reply:
I've just checked it and found that we don't need to save the digest here because it'll be overwritten here `pkg/skaffold/build/cache/details.go:121`.  
removed this part and pushed.